ERROR at //build/config/BUILDCONFIG.gn:533:5: Dependency not allowed.
    target(_target_type, target_name) {
    ^----------------------------------
The item //third_party/webrtc/pc:media_protocol_names
can not depend on //third_party/abseil-cpp/absl/strings:string_view
because it is not in //third_party/abseil-cpp/absl/strings:string_view's visibility list: [
  //third_party/abseil-cpp/*
  //third_party/googletest/*
  //third_party/webrtc/rtc_tools:frame_analyzer
  //third_party/webrtc/rtc_tools:rtp_generator
  //third_party/webrtc/rtc_tools:video_replay
  //chromecast/internal/*
  //libassistant/*
  //components/optimization_guide/internal/*
]


Absl used to have absl_deps, but that got removed with:
commit 99c519b3fdac0cbedaecead04a841e89a5f0f69b
Author: Florent Castelli <orphis@webrtc.org>
Date:   Thu May 23 11:21:53 2024 +0000

    Mass removal of absl_deps in all BUILD.gn files

Newer gn is fine with s/absl_deps/deps/, but bookworm's gn
requires manually changing the visibility of all the absl
libs.

--- a/third_party/abseil-cpp/absl/algorithm/BUILD.gn
+++ b/third_party/abseil-cpp/absl/algorithm/BUILD.gn
@@ -6,11 +6,13 @@ import("//third_party/abseil-cpp/absl.gn
 
 absl_source_set("algorithm") {
   public = [ "algorithm.h" ]
+  visibility = [ "*" ]
   deps = [ "//third_party/abseil-cpp/absl/base:config" ]
 }
 
 absl_source_set("container") {
   public = [ "container.h" ]
+  visibility = [ "*" ]
   deps = [
     ":algorithm",
     "//third_party/abseil-cpp/absl/base:config",
--- a/third_party/abseil-cpp/absl/base/BUILD.gn
+++ b/third_party/abseil-cpp/absl/base/BUILD.gn
@@ -40,6 +40,7 @@ absl_source_set("log_severity") {
 
 absl_source_set("no_destructor") {
   public = [ "no_destructor.h" ]
+  visibility = [ "*" ]
   deps = [
     ":config",
     ":nullability",
@@ -48,6 +49,7 @@ absl_source_set("no_destructor") {
 
 absl_source_set("nullability") {
   public = [ "nullability.h" ]
+  visibility = [ "*" ]
   deps = [ ":config" ]
 }
 
@@ -99,6 +101,7 @@ absl_source_set("config") {
     "options.h",
     "policy_checks.h",
   ]
+  visibility = [ "*" ]
 }
 
 absl_source_set("cycleclock_internal") {
@@ -137,6 +140,7 @@ absl_source_set("core_headers") {
     "port.h",
     "thread_annotations.h",
   ]
+  visibility = [ "*" ]
   deps = [ ":config" ]
 }
 
--- a/third_party/abseil-cpp/absl/cleanup/BUILD.gn
+++ b/third_party/abseil-cpp/absl/cleanup/BUILD.gn
@@ -15,6 +15,7 @@ absl_source_set("cleanup_internal") {
 
 absl_source_set("cleanup") {
   public = [ "cleanup.h" ]
+  visibility = [ "*" ]
   deps = [
     ":cleanup_internal",
     "//third_party/abseil-cpp/absl/base:config",
--- a/third_party/abseil-cpp/absl/container/BUILD.gn
+++ b/third_party/abseil-cpp/absl/container/BUILD.gn
@@ -63,6 +63,7 @@ absl_source_set("inlined_vector_internal
 
 absl_source_set("inlined_vector") {
   public = [ "inlined_vector.h" ]
+  visibility = [ "*" ]
   deps = [
     ":inlined_vector_internal",
     "//third_party/abseil-cpp/absl/algorithm",
@@ -112,6 +113,7 @@ absl_source_set("test_instance_tracker")
 
 absl_source_set("flat_hash_map") {
   public = [ "flat_hash_map.h" ]
+  visibility = [ "*" ]
   deps = [
     ":container_memory",
     ":hash_container_defaults",
@@ -141,6 +143,7 @@ absl_test("flat_hash_map_test") {
 
 absl_source_set("flat_hash_set") {
   public = [ "flat_hash_set.h" ]
+  visibility = [ "*" ]
   deps = [
     ":container_memory",
     ":hash_container_defaults",
--- a/third_party/abseil-cpp/absl/crc/BUILD.gn
+++ b/third_party/abseil-cpp/absl/crc/BUILD.gn
@@ -50,6 +50,7 @@ absl_source_set("crc32c") {
     "internal/crc_memcpy_x86_arm_combined.cc",
     "internal/crc_non_temporal_memcpy.cc",
   ]
+  visibility = [ "*" ]
   deps = [
     ":cpu_detect",
     ":crc_internal",
--- a/third_party/abseil-cpp/absl/flags/BUILD.gn
+++ b/third_party/abseil-cpp/absl/flags/BUILD.gn
@@ -164,6 +164,7 @@ absl_source_set("flag") {
     "declare.h",
     "flag.h",
   ]
+  visibility = [ "*" ]
   deps = [
     ":commandlineflag",
     ":config",
@@ -201,6 +202,7 @@ absl_source_set("usage_internal") {
 absl_source_set("usage") {
   sources = [ "usage.cc" ]
   public = [ "usage.h" ]
+  visibility = [ "*" ]
   deps = [
     ":usage_internal",
     "//third_party/abseil-cpp/absl/base:config",
@@ -217,6 +219,7 @@ absl_source_set("parse") {
     "internal/parse.h",
     "parse.h",
   ]
+  visibility = [ "*" ]
   deps = [
     ":commandlineflag",
     ":commandlineflag_internal",
--- a/third_party/abseil-cpp/absl/functional/BUILD.gn
+++ b/third_party/abseil-cpp/absl/functional/BUILD.gn
@@ -7,6 +7,7 @@ import("//third_party/abseil-cpp/absl.gn
 absl_source_set("any_invocable") {
   sources = [ "internal/any_invocable.h" ]
   public = [ "any_invocable.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/base:core_headers",
@@ -34,6 +35,7 @@ absl_test("any_invocable_test") {
 absl_source_set("bind_front") {
   sources = [ "internal/front_binder.h" ]
   public = [ "bind_front.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/container:compressed_tuple",
     "//third_party/abseil-cpp/absl/meta:type_traits",
@@ -67,6 +69,7 @@ absl_test("function_ref_test") {
 
 absl_source_set("overload") {
   public = [ "overload.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/meta:type_traits",
--- a/third_party/abseil-cpp/absl/memory/BUILD.gn
+++ b/third_party/abseil-cpp/absl/memory/BUILD.gn
@@ -6,6 +6,7 @@ import("//third_party/abseil-cpp/absl.gn
 
 absl_source_set("memory") {
   public = [ "memory.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:core_headers",
     "//third_party/abseil-cpp/absl/meta:type_traits",
--- a/third_party/abseil-cpp/absl/meta/BUILD.gn
+++ b/third_party/abseil-cpp/absl/meta/BUILD.gn
@@ -29,6 +29,7 @@ absl_test("requires_test") {
 
 absl_source_set("type_traits") {
   public = [ "type_traits.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/base:core_headers",
--- a/third_party/abseil-cpp/absl/numeric/BUILD.gn
+++ b/third_party/abseil-cpp/absl/numeric/BUILD.gn
@@ -9,6 +9,7 @@ absl_source_set("bits") {
     "bits.h",
     "internal/bits.h",
   ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/base:core_headers",
--- a/third_party/abseil-cpp/absl/strings/BUILD.gn
+++ b/third_party/abseil-cpp/absl/strings/BUILD.gn
@@ -7,6 +7,7 @@ import("//third_party/abseil-cpp/absl.gn
 
 absl_source_set("string_view") {
   sources = [ "string_view.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/base:core_headers",
@@ -54,6 +55,7 @@ absl_source_set("strings") {
     "strip.h",
     "substitute.h",
   ]
+  visibility = [ "*" ]
   public_deps = [
     # string_view.h was once part of :strings, so string_view.h is
     # re-exported for backwards compatibility.
@@ -145,6 +147,7 @@ absl_test("append_and_overwrite_test") {
 
 absl_source_set("str_format") {
   public = [ "str_format.h" ]
+  visibility = [ "*" ]
   deps = [
     ":str_format_internal",
     ":string_view",
--- a/third_party/abseil-cpp/absl/types/BUILD.gn
+++ b/third_party/abseil-cpp/absl/types/BUILD.gn
@@ -36,6 +36,7 @@ absl_test("source_location_test") {
 absl_source_set("span") {
   public = [ "span.h" ]
   sources = [ "internal/span.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/algorithm",
     "//third_party/abseil-cpp/absl/base:config",
@@ -82,6 +83,7 @@ absl_source_set("any_span") {
 
 absl_source_set("optional") {
   public = [ "optional.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/utility",
@@ -90,6 +92,7 @@ absl_source_set("optional") {
 
 absl_source_set("variant") {
   public = [ "variant.h" ]
+  visibility = [ "*" ]
   deps = [
     "//third_party/abseil-cpp/absl/base:config",
     "//third_party/abseil-cpp/absl/utility",
